go/exocom/src/connection/nested_service_mapping.go

```diff
@@ -42,6 +42,20 @@ func (n NestedServiceMapping) DeleteForService(key1 string, service *Service) {
 	}
 }
 
+// DeleteForActivityID deletes entries of a given service that contain activityID
+func (n NestedServiceMapping) DeleteForActivityID(activityID string) {
```

Comment:
This struct doesn't know its holder activityIds. Can we make this more generic? Something like `DeleteNestedKey`. We could also add some unit tests for this function since we don't have a specific feature test for it